cvmfs/catalog.h

```diff
@@ -231,6 +231,7 @@ class Catalog : public SingleCopy {
   Counters& GetCounters() { return counters_; }
 
   inline const CatalogDatabase &database() const { return *database_; }
+  inline       CatalogDatabase &database()       { return *database_; }
```

Comment:
Would be better to return a pointer.  What comes back from this function has value syntax but pointer semantics.